plugin.js

```diff
@@ -10,16 +10,20 @@ function factory (options) {
     errorResponse (err) {
       return { error: err.message }
     },
+    allowAnonymous: false,
     contentType: undefined,
     bearerType: 'Bearer'
   }
   const _options = { ...defaultOptions, ...options || {} }
   if (_options.keys instanceof Set) _options.keys = Array.from(_options.keys)
-  const { keys, errorResponse, contentType, bearerType, auth } = _options
+  const { keys, errorResponse, contentType, bearerType, auth, allowAnonymous } = _options
 
   function bearerAuthHook (fastifyReq, fastifyRes, next) {
     const header = fastifyReq.raw.headers.authorization
     if (!header) {
+      if (allowAnonymous) {
+        return next()
+      }
```

Comment:
It seems to me that this just disabled the plugin. I think this is better solved by using Fastify's encapsulation context support.

Reply:
I wouldn't say disables the plugin. But skips the validation if the Authorization header is not present

Reply:
Then I do not understand why the plugin would be registered at all. If you can skip the authorization by omitting the header, there's no reason to add the plugin to the context.